File: 3. Scripts/ruby/pa-test2.rb
require 'selenium-webdriver'
require 'rspec'
require_relative "singup_page.rb"
require_relative "user_page.rb"


timestemp = Time.now.to_i
email = "test#{timestemp}@test.me"
password = 'Testic123'
password2 = 'Testic123'
success_reg_text = "Hvala na registraciji!"

# Test for registration at polovni automoboili
describe 'Polovni automobili application:' do
  describe 'signup to the polovni automobili application' do
    it 'confirm that a user can successfully signup' do
      @driver = Selenium::WebDriver.for :firefox
      # Go to sing-in form
      @driver.navigate.to 'https://www.polovniautomobili.com/registracija'

      # Fill out form
      signup = SignupPage.new(@driver)
      signup.enter_email_address(email)
      signup.enter_password(password)
      signup.enter_password2(password2)
      signup.click_rules
      signup.click_easysale
      signup.click_easybuy
      signup.click_login

      sleep 1

      # Confirm user is singed up successufly
      user = UserPage.new(@driver)
      success_reg = user.user_success
      expect(success_reg).to eq(success_reg_text)
      
      @driver.quit


    end
  end
end

File: 3. Scripts/javascript/pa-test2.js
const { Builder } = require('selenium-webdriver');
const assert = require('assert');
const SignupPage = require('./signup_page');
const UserPage = require('./user_page');

const timestamp = Math.floor(Date.now() / 1000);
const email = `test${timestamp}@test.me`;
const password = 'Testic123';
const password2 = 'Testic123';
const successRegText = 'Hvala na registraciji!';

describe('Polovni automobili application:', function () {
  describe('signup to the polovni automobili application', function () {
    it('confirm that a user can successfully signup', async function () {
      const driver = await new Builder().forBrowser('firefox').build();

      // Go to sign-in form
      await driver.get('https://www.polovniautomobili.com/registracija');

      // Fill out form
      const signup = new SignupPage(driver);
      await signup.enterEmailAddress(email);
      await signup.enterPassword(password);
      await signup.enterPassword2(password2);
      await signup.clickRules();
      await signup.clickEasySale();
      await signup.clickEasyBuy();
      await signup.clickLogin();

      await driver.sleep(1000);

      // Confirm user is signed up successfully
      const user = new UserPage(driver);
      const successReg = await user.userSuccess();
      assert.strictEqual(successReg, successRegText);

      await driver.quit();
    });
  });
});